### PR TITLE
Fix artifacts upload

### PR DESCRIPTION
**Pull request by viktorb-ledger**
Replaces libcore artifact version from "4.1.1-SNAPSHOT" to 
- for releases: ${LIB_VERSION} (from CMakeLists.txt)
- for pushes: ${LIB_VERSION}-${latest-reachable-tag}-${N}-${commit-hash}-SNAPSHOT"
- for pull requests: ${LIB_VERSION}-${BRANCH_NAME}-${BRANCH_LENGTH}-${commit_hash}-SNAPSHOT

where ${N} - number of commits from last reachable tag.

### CODE REVIEW

**Review by twilgenbus-ledger — COMMENTED** [review]
Very neat idea to have a monotonic counter from the latest tag. (yet still include sha to protect from collisions with other branches)

According to the [latest CI run](https://github.com/LedgerHQ/lib-ledger-core/runs/4627930568?check_suite_focus=true#step:8:1800
), there are things that confuse me:
- the version appears twice, due to the format being $VERSION-$LATEST_TAG and because we signal the version in the tag (i.e 4.1.1-4.1.0-rc3). Not sure if intended/if we have a better solution?
- the ${commit-hash} (i.e g9e3662a9d) does not fit current commit hash
caf94d366cf676b1f7cdca16e52c99c4253ce3ab, I am misunderstanding something?

**Comment by jcoatelen-ledger** [comment]
Can you make sure to have all tests green on the branch? Looks like this branch is missing some configuration fixes already pushed on `release/4.1.0`.

**Comment by viktorb-ledger** [comment]
> the version appears twice, due to the format being $VERSION-$LATEST_TAG and because we signal the version in the tag (i.e 4.1.1-4.1.0-rc3). Not sure if intended/if we have a better solution?

In many practical cases these two mentioned versions would be different:
- the first one comes from CMakeLists.txt and indicates the future release we are working on
- the second one comes from latest git tag and indicates the previously released / tagged version
The resulting string is redundant but easily understandable: `${version}-${latest-reachable-tag}-${N}-${commit-hash}` or `${version}` for releases. It's possible to remove duplicated part of latest tag but it would be more difficult to find out where the latest tag starts.

**Comment by viktorb-ledger** [comment]
> the ${commit-hash} (i.e g9e3662a9d) does not fit current commit hash caf94d3, I am misunderstanding something?

https://github.com/LedgerHQ/lib-ledger-core/commit/9e3662a9d - is a merge commit created by GitHub. It merges the head of the target branch (release/4.1.0 in our case) and the head of pull request ( caf94d3 ) . Yes, this isn't a very informative commit hash in our case.

File: nix/scripts/export_libcore_version.sh [hunk 1]
```diff
@@ -26,8 +27,9 @@ if [[ "${GITHUB_EVENT_NAME:-NO}" == "release" ]]; then
 elif [[ "${GITHUB_EVENT_NAME:-NO}" == "push" ]]; then
     printf "\n${GITHUB_EVENT_NAME:-NO} is \"push\", creating a snapshot\n"
     echo "::set-output name=deploy_dynlibs::NO"
-    echo "::set-output name=jar_version::${LIB_VERSION}-SNAPSHOT"
+    echo "::set-output name=jar_version::${LIB_VERSION}-${LIBCORE_GIT_DESCRIBE}-SNAPSHOT"
 elif [[ "${GITHUB_EVENT_NAME:-NO}" == "pull_request" ]]; then
-    printf "Github event matched the \"pull_request\" type\n"
-    printf "No upload being done, cause this is a pull request from a forked repository\n"
+    printf "\n${GITHUB_EVENT_NAME:-NO} is \"pull_request\", creating a snapshot\n"
```
[the remark this answers is not in the document]
Comment (viktorb-ledger) [reply nix/scripts/export_libcore_version.sh]: Done.

File: .github/workflows/nix_release_builds.yml [hunk 2]
```diff
@@ -24,6 +24,8 @@ jobs:
       jar_version: ${{ steps.lib_version.outputs.jar_version }}
     steps:
       - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (viktorb-ledger) [reply .github/workflows/nix_release_builds.yml]: Done

**Comment by viktorb-ledger** [comment]
> Can you make sure to have all tests green on the branch? Looks like this branch is missing some configuration fixes already pushed on `release/4.1.0`.

Failing tests in this branch are still random: they depend on network failures.
Which fixes are missing?

**Review by jcoatelen-ledger — APPROVED** [review]

**Pull request merged**